client/checkout/classic/upe.js

```diff
@@ -371,13 +392,18 @@ jQuery( function ( $ ) {
 		upeElement.mount( upeDOMElement );
 		unblockUI( $upeContainer );
 		upeElement.on( 'change', ( event ) => {
-			const selectedUPEPaymentType = event.value.type;
+			let selectedUPEPaymentType = event.value.type;
+			if ( 'card' === selectedUPEPaymentType ) {
```

Comment:
Irritatingly, despite the payment element containing both `link` and `card` in its `payment_method_types` parameter, this result will always return as `card`. Go figure.